ls1intum/Artemis
# [Refactor] improved ExamResource and Authorization in ExamAccessService

### Checklist

### Motivation and Context



The ExamResource was very big and used a lot of old methods we slowly want to get rid of, the ExamAccessService used a method for every possible authentication case and every possible Role. This needed to be made easier.

### Description


Unified Authorization in ExamAccessService, adapted ExamResource to existing Guidelines for Rest-Endpoints

### Steps for Testing

For most of the Endpoints to be tested, Instructor rights are needed.
During testing, keep your console open and especially check the network tab for any errors out of the ordinary!
1. Log in to Artemis  
2. create an exam 
3. after creation, change something about the exam, e.g. title
4. look at the statistics of an exam 
5. change the order of exercise groups in an exam
6. check the assessment dashboard of a finished exam 
7. make an exam test run and check the statistics and assessment dashboard for this test run afterward 
8. delete an exam 
9. add a single user to an exam, then add all course students to the exam 
10. generate all student exams, then add another student and generate the missing individual exams 
11. unlock all repositories for an exam (done in the same view as creating student exams) 
12. lock all repositories for an exam 
13. remove a single user from the exam 
14. delete all remaining students from the exam (by pressing the remove all button)
15. start an exam as a student
16. archive an exam 
17. download the archive of exams 
18. look at the list of upcoming exams (need admin for that, go to server administration > Upcoming exams)

Use the checkboxes after the action items to track your own progress while testing and to let others know what you tested

### Review Progress





- Code Review
- Manual Tests

### Test Coverage



ExamResource: 97%
